File: scripts/vyatta-validate-type.pl
#!/usr/bin/perl

# Author: An-Cheng Huang <ancheng@vyatta.com>
# Date: 2007
# Description: script to validate types

# **** License ****
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License version 2 as
# published by the Free Software Foundation.
# 
# This program is distributed in the hope that it will be useful, but
# WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# General Public License for more details.
# 
# This code was originally developed by Vyatta, Inc.
# Portions created by Vyatta are Copyright (C) 2006, 2007, 2008 Vyatta, Inc.
# All Rights Reserved.
# **** End License ****

use strict;
use lib "/opt/vyatta/share/perl5/";
use Vyatta::TypeChecker;

# validate a value of a specific type
if ($#ARGV < 1) {
  print "usage: vyatta-validate-type.pl [-q] <type> <value>\n";
  exit 1;
}

my $quiet = undef;
if ($ARGV[0] eq '-q') {
  shift;
  $quiet = 1;
}

exit 0 if (Vyatta::TypeChecker::validateType($ARGV[0], $ARGV[1], $quiet));
exit 1;

File: utils/vyatta-validate-type.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "type_check.h"

int
main (int argc, char *argv[])
{
  if (argc < 2) {
    printf("usage: vyatta-validate-type [-q] <type> <value>\n");
    return 1;
  }
  if (strcmp(argv[1], "-q") == 0){
    if (validateType(argv[2], argv[3], 1)) {
      return 0; 
    }
    return 1;
  } 
  if (validateType(argv[1], argv[2], 0)) {
    return 0; 
  }
  return 1;
}
